### PR TITLE
Add UT cases for Preempt Action

### PR DESCRIPTION
**Pull request by thandayuthapani**
**What this PR does / why we need it**:
Added UT Cases for Preempt Action

### CODE REVIEW

**Comment by thandayuthapani** [comment]
Hi @TommyLike , @hzxuzhonghu  can you please review this PR?

File: pkg/scheduler/actions/preempt/preempt_test.go [hunk 1]
```diff
@@ -18,15 +18,185 @@ package preempt
...
+	tests := []struct {
+		name      string
+		podGroups []*kbv1.PodGroup
+		pods      []*v1.Pod
+		nodes     []*v1.Node
+		queues    []*kbv1.Queue
+		expected  int
+	}{
+		{
+			name: "one Job with two Pods on one node",
```
Comment (hzxuzhonghu) [inline pkg/scheduler/actions/preempt/preempt_test.go:51]: Confused about the name,  I am not familiar with the scheduler part, though.

Comment (thandayuthapani) [reply pkg/scheduler/actions/preempt/preempt_test.go:51]: Those are just test case names.  Have given a one line description about what testcase is trying to do.

File: pkg/scheduler/util/test_utils.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		},
+		Status: v1.NodeStatus{
+			Capacity:    alloc,
+			Allocatable: alloc,
+		},
+	}
+}
+
+// BuildPod builts Pod object
+func BuildPod(ns, n, nn string, p v1.PodPhase, req v1.ResourceList, groupName string, labels map[string]string, selector map[string]string) *v1.Pod {
```
Comment (hzxuzhonghu) [inline pkg/scheduler/util/test_utils.go]: prefer `namespace, name, nodeName `

File: pkg/scheduler/actions/preempt/preempt_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/scheduler/actions/preempt/preempt_test.go:51]: Yes, i know. Just curious why one job and where is the job? Overall just a nit.

File: pkg/scheduler/actions/preempt/preempt_test.go [hunk 3]
```diff
@@ -18,15 +18,185 @@ package preempt
...
+			podGroups: []*kbv1.PodGroup{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pg1",
+						Namespace: "c1",
+					},
+					Spec: kbv1.PodGroupSpec{
+						Queue: "c1",
```
Comment (hzxuzhonghu) [inline pkg/scheduler/actions/preempt/preempt_test.go]: nit: The queue name is `c1`  and the namespace is also called `c1`

File: pkg/scheduler/actions/preempt/preempt_test.go (hunk 1 above)
Comment (thandayuthapani) [reply pkg/scheduler/actions/preempt/preempt_test.go:51]: One podGroup is created in that test case, so all pods in a PodGroup is handled as all tasks under a Job in code.  So here only one PodGroup is created, which is internally(in code) one job in that queue.  That is why termed the test case name as one job.

File: pkg/scheduler/util/test_utils.go (hunk 2 above)
Comment (thandayuthapani) [reply pkg/scheduler/util/test_utils.go]: Done

File: pkg/scheduler/actions/preempt/preempt_test.go (hunk 3 above)
Comment (thandayuthapani) [reply pkg/scheduler/actions/preempt/preempt_test.go]: Done.  Have changed queue name to q1

File: pkg/scheduler/actions/preempt/preempt_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/scheduler/actions/preempt/preempt_test.go:51]: ok

**Pull request merged**